src/client/Microsoft.Identity.Client/Http/HttpManager.cs

```diff
@@ -124,10 +125,9 @@ private HttpRequestMessage CreateRequestMessage(Uri endpoint, IDictionary<string
                     MsalErrorMessage.HttpRequestUnsuccessful,
                     (int)response.StatusCode, response.StatusCode));
 
-                if ((int)response.StatusCode >= 500 && (int)response.StatusCode < 600)
-                {
-                    isRetryable = true;
-                }
+
+                is5xxError = (int)response.StatusCode >= 500 && (int)response.StatusCode < 600;
+                isRetryable = is5xxError && !HasRetryAfterHeader(response);
```

Comment:
Updated our HTTP logic around resilience. 

Old rule: "retry once each requests that returns HTTP Status 5xx "
New rule: "retry once each requests that returns HTTP Status 5xx, **except if a Retry-After header is present**"

Reply:
when would the retry-after header present?

---
In reply to: [412362816](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/1774#discussion_r412362816) [](ancestors = 412362816)

Reply:
Server can send this, typically associated with HTTP Status 429 (Too Many Requests) or 503 (Unavailable). It's a mechanism used by server to tell clients to back off.